keras/preprocessing/sequence.py

```diff
@@ -216,3 +218,93 @@ def _remove_long_seq(maxlen, seq, label):
             new_seq.append(x)
             new_label.append(y)
     return new_seq, new_label
+
+
+class TimeSeriesSequence(Sequence):
+    """Utility class for generating batches of temporal data.
+
+    This class takes in a sequence of data-points gathered at 
+    equal intervals, along with time series parameters such as 
+    stride, delay etc., to produce batches for training/validation.
+
+    # Arguments
+        data: indexable generator (such as list) containing
+            consecutive data points.
+        targets: targets corresponding to instances in `data`.
+            Should have same length as data.
+        lookback: length of history to consider for prediction.
+            data[i-lookback : i+1] is used as input.
+        delay: how further in the future we are looking to predict.
+            targets[i + delay] becomes the target for
+            data[i-lookback : i+1]
+        sampling_rate: period between successive individual data
+            points. For rate `r`, points
+            data[i], data[i-r], ... data[i - lookback] are used as input.
+        stride: period between successive sub-sequences used for
+            training. For stride s, consecutive training samples would
+            be centered around data[i], data[i+s], data[i+2*s], ...
+        start_index, end_index: delimiters for sub-sets of data & targets
+            to be used for batch generation. Useful to reserve part of
+            data for validation.
+        shuffle: whether to shuffle training samples, or draw them in
+            chronological order.
+        reverse: if true, input points are provided in reverse order.
+        batch_size: size of successful batches (except maybe last one)
+    
+    # Returns
+        A `Sequence` instance.
+    """
+    def __init__(self, data, targets,
+                 lookback,
+                 delay,
+                 sampling_rate=1,
+                 stride=1,
+                 start_index=0,
+                 end_index=None,
+                 shuffle=False,
+                 reverse=False,
+                 batch_size=128):
+        self.data = data
+        self.targets = targets
+        if not hasattr(self.targets[0], "__len__"):
+            self.output_dim = 1
+        else:
+            self.output_dim = len(self.targets[0])
+        self.lookback = lookback
+        self.delay = delay
+        self.sampling_rate = sampling_rate
+        self.stride = stride
+        self.start_index = start_index + lookback
+        if end_index == None or end_index + delay > len(data) - 1:
+            end_index = len(data) - delay - 1
+        self.end_index = end_index
+        self.shuffle = shuffle
+        self.reverse = reverse
+        self.batch_size = batch_size
+
+    def __len__(self):
+        return np.ceil(
+                (self.end_index - self.start_index) / 
+                (self.batch_size * self.stride))
+
+    def __getitem__(self, index):
+        if self.shuffle:
```

Comment:
This will shuffle the data for each batch. Is it wanted?
I thought `shuffle` would just shuffle the dataset in the `on_epoch_end` callback

Reply:
Is shuffling a big plus for a time-series generator anyway? 
We could 'shift' the beginning cursor to a different point at `on_epoch_end` as you suggest, so that the batches in the next epoch are different.